draft-gpew-priv-ppm.md

```diff
@@ -672,247 +677,377 @@ independently.
-parameters associated with task `task_id`. Then, using the procedure in
-{{batch-parameter-validation}}, it ensures that the request meets the
-requirements of the batch parameters. It also computes a checksum based on its
-view of the output shares included in the batch window, and checks that the
-`report_count` and `checksum` included in the request match its computed
-values. If so, it aggregates all valid output shares that fall in the batch
-interval into an aggregate share. The response contains an opaque,
-VDAF-specific message:
+The helper continues with preparation for a report share by combining the leader's input
+message in `AggregateContinueReq` and its current preparation state (`prep_state`). This
+step yields one of three outputs:
+
+1. An error, in which case the input report share is marked as invalid.
+1. An output share, in which case the helper stores the output share for future collection
+   as described in {{collect-flow}}.
+1. An updated VDAF state and preparation message, denoted `(prep_state, prep_msg)`.
+
+To carry out this step, for each PrepareStep in AggregateContinueReq.prepare_shares received
+from the leader, the helper performs the following check to determine if the report share
+should continue being prepared.
+
+* If failed, then mark the report as failed and reply with a failed PrepareStep
+  to the leader.
+* If finished, then mark the report as finished and reply with a finished PrepareStep
+  to the leader. The helper then stores the output share and awaits for collection;
+  see {{collect-flow}}.
+
+Otherwise, preparation continues. In this case, the helper computes its updated state
+and output message as follows:
 
 ~~~
-struct {
-  opaque message<0..2^16-1>; // VDAF message
-} AggregateShare;
+out = VDAF.prep_next(prep_state, inbound)
 ~~~
 
-Next, the helper encrypts the aggregate share `agg_share` under the collector's
-public key as follows:
+where `inbound` is the previous VDAF preapre message sent by the leader and `prep_state` is
+its current preparation state. If this operation fails, then the helper fails
+with error `vdaf-prep-error`. Otherwise, it interprets `out` as follows. If this
+is the last round of VDAF preparation phase, then `out` is the helper's output
+share, in which case the helper stores the output share for future collection.
+Otherwise, the helper interpets `out` as the tuple `(new_state, prep_msg)`, where
+`new_state` is its updated preparation state and `prep_msg` is its next VDAF
+message.
+
+This output message for each report in AggregateContinueReq.prepare_shares is then sent
+to the leader in an AggregateContinueResp message, structured as follows:
 
 ~~~
-enc, context = SetupBaseS(pk, task_id,
-                          "ppm-00 aggregate share" || 0x03 || 0x00)
-encrypted_agg_share = context.Seal(batch_interval, agg_share)
+struct {
+  opaque helper_state<0..2^16>;
+  PrepareStep prepare_shares<1..2^16-1>;
```

Comment:
nit: for consistency with similar abbreviation elsewhere
```suggestion
  PrepareStep prep_shares<1..2^16-1>;
```

Reply:
Keeping this as-is for now. We can bash the name in a followup change.